model_analyzer/config/generate/run_config_generator_factory.py

```diff
@@ -112,26 +117,28 @@ def _create_quick_plus_concurrency_sweep_run_config_generator(
         command_config: ConfigCommandProfile, gpus: List[GPUDevice],
         models: List[ModelProfileSpec],
         ensemble_composing_models: Dict[str, List[ModelProfileSpec]],
-        client: TritonClient, result_manager: ResultManager,
+        bls_composing_models: List[ModelProfileSpec], client: TritonClient,
+        result_manager: ResultManager,
         model_variant_name_manager: ModelVariantNameManager
     ) -> ConfigGeneratorInterface:
         search_config = RunConfigGeneratorFactory._create_search_config(
-            models, ensemble_composing_models)
+            models, ensemble_composing_models, bls_composing_models)
         return QuickPlusConcurrencySweepRunConfigGenerator(
             search_config=search_config,
             config=command_config,
             gpus=gpus,
             models=models,
             ensemble_composing_models=ensemble_composing_models,
+            bls_composing_models=bls_composing_models,
             client=client,
             result_manager=result_manager,
             model_variant_name_manager=model_variant_name_manager)
 
     @staticmethod
     def _create_search_config(
-        models: List[ModelProfileSpec],
-        ensemble_composing_models: Dict[str, List[ModelProfileSpec]]
-    ) -> SearchConfig:
+            models: List[ModelProfileSpec],
+            ensemble_composing_models: Dict[str, List[ModelProfileSpec]],
```

Comment:
This feels like it is getting less and less clean. We are passing in a bunch of optional dicts/lists, and the contents of them imply whether or not the feature is enabled? It seems that would be really confusing for someone who doesn't have intimate knowledge of the code base.

Are there ways to refactor or clean all of this up? Can there be a thin layer of prep for normal vs multi vs ensemble vs bls, and have all of the underlying code be shared? I think that is sort of what is already happening, but just want to take a step back and think if there are better ways to organize this code. It's hard for me to make those calls just looking at the PR diffs.